### PR TITLE
util-linux: update to v2.37.4 to fix CVE-2021-3995

### PR DESCRIPTION
**Pull request by 0xba1a**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Upgrade util-linux package to version 2.37.4

###### Change Log  

- Upgrade util-linux package to version 2.37.4
- Build util-linux without audit (older version also didn't have audit support)
- Avoid CVE-2021-3995 which was found in older version

###### Does this affect the toolchain?  

**YES**

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2021-3995

###### Test Methodology

- Local package build

### CODE REVIEW

**Comment by 0xba1a** [comment]
BuddyBuild

- AMD64: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=247081&view=results
- ARM64: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=247082&view=results

Complete build
- https://dev.azure.com/mariner-org/mariner/_build/results?buildId=247083&view=results

File: SPECS/util-linux/util-linux.spec [hunk 1]
```diff
@@ -1,34 +1,38 @@
 Summary:        Utilities for file systems, consoles, partitions, and messages
 Name:           util-linux
-Version:        2.32.1
```
[the remark this answers is not in the document]
Comment (0xba1a) [reply SPECS/util-linux/util-linux.spec:3]: All the versions upto 2.37.3 are vulnerable
image

File: SPECS/util-linux/util-linux.spec [hunk 2]
```diff
@@ -1,34 +1,38 @@
...
+Source1:        runuser
+Source2:        runuser-l
+Source3:        su
+BuildRequires:  libcap-ng-devel
 BuildRequires:  libselinux-devel
 BuildRequires:  ncurses-devel
+BuildRequires:  pam-devel
+Requires:       %{name}-libs = %{version}-%{release}
+# util-linux contains su, which conflicts with shadow-utils < 4.9-10 that also contained su
+Conflicts:      shadow-utils < 4.9-10
```
[the remark this answers is not in the document]
Comment (0xba1a) [reply SPECS/util-linux/util-linux.spec:20]: I took util-linux form the `main` branch.  For 2.0 `su` tool is provided by `util-linux`. But in 1.0 it is provided by `shadow-utils`.

To maintain the same, let me remove `su` from `util-linux` and remove the conflict as well.

File: SPECS/util-linux/util-linux.spec (hunk 1 above)
[the remark this answers is not in the document]
Comment (oliviacrain) [reply SPECS/util-linux/util-linux.spec:3]: Are we not able to backport the patch?

[the remark this answers is not in the document]
Comment (oliviacrain) [reply SPECS/util-linux/util-linux.spec:3]: Also, we should try to exploit this vulnerability ourselves, to see if NVD's data is correct. [Ubuntu is saying](https://ubuntu.com/security/CVE-2021-3995) that this was introduced in v2.34.

**Review by oliviacrain — CHANGES_REQUESTED** [review]
Blocking- would like confirmation that we're actually vulnerable to this bug, please see above comment.

**Review by jslobodzian — CHANGES_REQUESTED** [review]
The buddy build failed.  We should backport the change.  I think this is too large/risky for Mariner 1.0 at the moment.

File: SPECS/util-linux/util-linux.spec (hunk 1 above)
[the remark this answers is not in the document]
Comment (0xba1a) [reply SPECS/util-linux/util-linux.spec:3]: @oliviacrain I tried applying the patch first. It's not applying. Some functions are not yet implemented in the older version. So manually updating the patch will not be a good idea.

Let me explore the seriousness of the vulnerability.

[the remark this answers is not in the document]
Comment (oliviacrain) [reply SPECS/util-linux/util-linux.spec:3]: I looked at the [initial posting of the vuln to openwall](https://www.openwall.com/lists/oss-security/2022/01/24/2). It lists the below commit as the initial introduction of the vulnerability.

https://github.com/util-linux/util-linux/commit/5fea669e9ef0a08804f72bb40f859f239f68c30a

This commit is introduced in v2.34, and we're on v2.32 here, so I think we can safely close this PR. We do need to ask NIST to correct the NVD database, though.

[the remark this answers is not in the document]
Comment (neha170) [reply SPECS/util-linux/util-linux.spec:3]: I had reached the same conclusion wrt when the vulnerability was introduced, and had sent a mail to NIST. Awaiting reply, closing this PR.

**Comment by neha170** [comment]
Closing this PR. Our version shouldn't be affected by this, as the offending code was introduced in a later version. Awaiting reply from NIST.

**Pull request closed without merging**